### ISSUE TITLE
MongoDB batch classes support

### ISSUE THREAD

**Issue by jenn0pal** [yiisoft/yii2-mongodb#33]
Will you be supporting [Batch Classes](http://php.net/manual/en/mongo.batch.php)?

The only batch method that I've seen is batchInsert() but there's no support for batchUpdate().

This will be more efficient than looping $collection->update()

```php
public static function syncStock()
    {
        $items = InventoryItems::getAllStocks();
        foreach ($items as $item) {
             self::getCollection()->update(
                ['sku' => $item['sku']],
                [
                    'sku' => $item['sku'],
                    'qty' => $item['qty'],
                    'store_id' => $item['store_id']
                ],
                ['upsert' => true]
            );
        }
        return Yii::getLogger()->getElapsedTime();
    }
/* Updating 5000 items resulted in 46.0376329422 */
```

```php
public static function  batchSyncStock()
    {
        $mc = new \MongoClient();
        $collection = $mc->selectCollection('oms_prod', 'product_stocks');

        $batch = new \MongoUpdateBatch($collection);
        $items = InventoryItems::getAllStocks();
        foreach ($items as $item) {
            $batch->add([
                'q' => ['sku' => $item['sku']],
                'u' => ['$set' => [
                    'sku' => $item['sku'],
                    'qty' => $item['qty'],
                    'store_id' => $item['store_id']
                ]],
                'upsert' => true,
            ]);
        }

        $batch->execute();
        return Yii::getLogger()->getElapsedTime();
    }
/* Updating 5000 items resulted in 18.461055994034 */
```

**Comment by klimov-paul** [comment]
Will be considered at 2.1

**Comment by klimov-paul** [comment]
Resolved via #116
